policies.go

```diff
@@ -464,7 +465,7 @@ func (t *tokenAwareHostPolicy) SetPartitioner(partitioner string) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
 
-	if t.partitioner != partitioner {
+	if t.partitioner == partitioner {
```

Comment:
I honestly don't understand this change. Looks it introduces a bug to me.

Reply:
Yeah this is a mistake, good catch.